### DISCUSSION TITLE
Multiple indexes with the same structure

### DISCUSSION THREAD

**Discussion by toooni** [schranz-search/schranz-search#266] [Q&A] [answered]
Hi

I am currently looking into schranz-search to replace our old elasticsearch/opensearch setup. In our current setup we have multiple indexes with the same structure:
- `package_de_chf`
- `package_fr_chf`
- `package_de_eur`
- `package_fr_eur`

I do think it makes sense to keep these indexes for performance reasons. How would you set this up in the schema folder?

**Comment by alexander-schranz** [comment, marked as answer]
There is not yet an out of the box solution for this but you should be able to reuse the field definition of another index to avoid duplicated schema definitions:

```php
$indexDeChf = require_once(__DIR__ . '/package_de_chf.php');

return new Index('package_fr_chf', $indexDeChf->fields);
```

**Reply by toooni** [reply]
Thanks. I've came to a similar solution. I'd be able to contribute if you have an idea how to make this better? Maybe a MultiIndex class? Or the first param of Index could also be an array of strings?

**Reply by alexander-schranz** [reply]
Revisiting this discussion at current state not yet sure if it will be part of the core but mostly I would go this way:

```php
return new Index('package_%locale%', [/* ... */]);
```

Then decorate the Schema Loader Service via:

```php
namespace CmsIg\Seal\Schema\Loader;

use CmsIg\Seal\Schema\Loader\LoaderInterface;

class LocaleMetadataLoader implements LoaderInterface
{
    /** @var iterable<string|Stringable> $locales */
    public function __construct(private LoaderInterface $schemaLoader, private array $locales) {}

    public function load(): Schema
    {
        $schema = $this->schemaLoader->load();
        
        /** @var array<string, Index> $indexes */
        $indexes = [];
        foreach ($schema->index as $key => $index) {
             foreach ($this->locales as $locale) {
                  $indexKey = str_replace('%locale%', $locale, $key);
                  $indexName = str_replace('%locale%', $locale, $index->name);
                  $indexes[$indexKey] = new Index($indexName, $index->fields);
             }
        }
        
        return new Schema($indexes);
    }
}

```
